app/main/views/brief_responses.py

```diff
@@ -0,0 +1,117 @@
+from flask import jsonify, abort, request, current_app
+from sqlalchemy.exc import IntegrityError, DataError
+
+from dmapiclient.audit import AuditTypes
+
+from .. import main
+from ...models import db, Brief, BriefResponse, AuditEvent
+from ...utils import (
+    get_json_from_request, json_has_required_keys,
+    pagination_links, get_valid_page_or_1, url_for,
+    validate_and_return_updater_request,
+)
+
+from ...service_utils import validate_and_return_supplier
+
+
+@main.route('/brief-responses', methods=['POST'])
+def create_brief_response():
+    json_payload = get_json_from_request()
+    updater_json = validate_and_return_updater_request()
+
+    json_has_required_keys(json_payload, ['briefResponses'])
+    brief_response_json = json_payload['briefResponses']
+    json_has_required_keys(brief_response_json, ['briefId', 'supplierId'])
+
+    try:
+        brief = Brief.query.get(brief_response_json['briefId'])
+    except DataError:
+        brief = None
+
+    if brief is None:
+        abort(400, "Invalid brief ID '{}'".format(brief_response_json['briefId']))
+
+    if brief.status != 'live':
+        abort(400, "Brief must be live")
+
+    if brief.framework.status != 'live':
+        abort(400, "Brief framework must be live")
```

Comment:
Could these also be handled by the `validate` method? They are invariants that make creating the `BriefResponse` invalid.